react/react1/week2/todo-list/src/components/App/App.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from "react";
+import { List } from "../List/List";
+import { Input } from "../Input/Input";
+import { Button } from "../Button/Button";
+import { Timer } from "../Timer/Timer";
+
+
+export default class App extends Component {
+  state = {
+    todos: [
+      { id: 1, description: "Get out of bed", done: false },
+      { id: 2, description: "Brush your teech", done: false },
+      { id: 3, description: "Eat breakfast", done: false },
+    ],
+    todoText: ""
+
+  };
+
+  onChangeInput = e => {
+    this.setState({ todoText: e.target.value });
+  };
+
+  onSubmitTodo = () => {
+    this.setState(({ todos, todoText }) => ({
+      todos: [...todos, { id: todos.length + 1, description: todoText, done: false }],
```

Comment:
There is a small bug here in certain cases by using the list's length to create a new id.
Let's say you have todos with 1,2 and 3 ids.
Then you add a new one which gets the id 4.
Then you decide to remove the todo which has id 3.
If you try to add new todo now, it will get the id 4 again since the length of the list is 3

So, you could either generate a random id, or you could use the last todo's id in the list and increment it

Reply:
done